### PR TITLE
add support for saving credentials with your api key

### PR DESCRIPTION
**Pull request by davecap**
- [x] Add new docs about saving credentials

### CODE REVIEW

**Comment by jsh2134** [comment]
not sure if this expected behavior but....CLI login works as we expect and creates the credentials file.

```
solvebio login --api-key=XXXX
You are logged-in to the "AAAA" domain as AAA.AAA@AAAA.com with role member.
You are now logged-in as  AAA.AAA@AAAA.com.

## Note: it does print twice ^^
```

But if I try to login from within the client it does not work

```
$ SOLVEBIO_API_KEY=invalid-key solvebio
SolveBio Python shell started.
Error: Invalid API key
Invalid credentials. You may not be logged-in.
No authentication credentials found.

# If I try and call login() it works as expected.

login()
Please enter your SolveBio credentials
Domain (e.g. <domain>.solvebio.com): q

# But if I try and pass an api_key it does not

login(api_key='VALID_KEY')

TypeError: login() got an unexpected keyword argument 'api_key'


In [3]: login
Out[3]: <function solvebio.cli.auth.login>
```

Also if I import solvebio and then login, it also does not create the credentials files
```
import solvebio
solvebio.login(api_key="REAL_KEY")
solvebio.User.retrieve()
# login works! but no credentials file created
```

**Comment by jsh2134** [comment]
cc @davecap

**Comment by davecap** [comment]
Thanks! I'll look it over again.

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-- '2.6'
 - '2.7'
 - '3.5'
 - '3.6'
   # - 'pypy'
 install:
-- pip install requests==2.10.0 flake8==2.6.2 nose mock tox
-- if [[ $TRAVIS_PYTHON_VERSION == '2.6' ]]; then pip install ordereddict unittest2;
-  fi
+- pip install -r requirements.txt
```
Comment (jsh2134) [inline .travis.yml]: can we rename this requirements_dev.txt or something that denotes it is for dev/testing?

File: solvebio/test/__init__.py [hunk 2]
```diff
@@ -1,20 +0,0 @@
-from __future__ import absolute_import
-import pkgutil
-import unittest
-
-# FIXME: redo how testing works.
```
Comment (jsh2134) [inline solvebio/test/__init__.py]: 👍

File: .travis.yml (hunk 1 above)
Comment (davecap) [reply .travis.yml]: Yep

**Comment by davecap** [comment]
(travis check didn't update on success, closing the PR and re-opening triggers a travis rebuild)

**Comment by davecap** [comment]
Docs updated: https://github.com/solvebio/solvebio-docs/pull/131

**Pull request merged**